internal/grpc/services/ocmshareprovider/ocmshareprovider.go

```diff
@@ -144,7 +149,31 @@ func (s *service) CreateOCMShare(ctx context.Context, req *ocm.CreateOCMShareReq
 		}, nil
 	}
 
-	share, err := s.sm.Share(ctx, req.ResourceId, req.Grant, name, req.RecipientMeshProvider, permissions, nil, "", ocm.Share_SHARE_TYPE_REGULAR)
+	// discover share type
+	sharetype := ocm.Share_SHARE_TYPE_REGULAR
+	protocol, ok := req.Opaque.Map["protocol"]
+	if ok {
+		var sp *shareProtocol
+		switch protocol.Decoder {
+		case "json":
+			err := json.Unmarshal(protocol.Value, &sp)
```

Comment:
You're not using sp anywhere. Why are you even reading it?

Reply:
The protocol value is read into it (line 159: `err := json.Unmarshal(protocol.Value, &sp)`). Then the sharetype is established from testing the protocol name (line 165: `if sp.Name == "datatx"`). I'll move line 156 to line 159.